docs/architecture/adr/0022-server-authorization.md

```diff
@@ -0,0 +1,162 @@
+  additional requirements
+- in organizational contexts, we use role-based authorization, where the user is assigned a role in
+  an organization and permission is granted or denied based on that role. This closely tracks
+  [how access control is presented to the user](https://bitwarden.com/help/user-types-access-control/).
+  However, the expansion of collections has complicated this simple model, and it now looks more
+  like
+  [attribute-based access control](https://en.wikipedia.org/wiki/Attribute-based_access_control)
+
+This logic is spread throughout controllers in the API layer, JWT claims accessed via
+`CurrentContext`, the core service layer, and database queries.
+
+### Requirements
+
+Our authorization requirements have increased in complexity, particularly with the release of
+collection management enhancements. Today, the outcome of an authorization decision in respect of an
+organization resource could depend on the user's role, their collection relationships, organization
+settings, provider relationships, and other factors.
+
+A solution to this problem should:
+
+- separate authorization logic from other concerns
+- centralize authorization logic in a single location or pattern as much as possible
+- separate authorization logic from authorization checks/enforcement
+- be reusable between endpoints that access the same resource
+- support a range of authorization logic (e.g. based on role, resource, relationships, etc)
+
+## Considered Options
+
+### ASP.NET Core resource-based authorization
+
+#### Summary
+
+[Resource-based authorization](https://learn.microsoft.com/en-us/aspnet/core/security/authorization/resourcebased?view=aspnetcore-8.0)
+makes authorization decisions based on the **resource** being accessed, the **user** accessing the
+resource, and the **operations** (actions) they wish to take on the resource.
+
+To define authorization logic, developers implement one or more `IAuthorizationHandler` classes for
+the resource being accessed.
+
+Authorization checks are handled by the default implementation of `IAuthorizationService`, which
+calls each authorization handler for the resource to find at least 1 that will authorize the action.
+
+#### Advantages
+
+- included in ASP.NET - standard C# code, no additional dependencies
+- already used by Secrets Manager and for some collection operations in Password Manager
+- handlers provide good encapsulation of authorization logic, separate to other concerns and the
+  `AuthorizationService` implementation itself
+- the fixed `AuthorizationService` interface enforces a consistent usage across our teams
+- teams can write and have code ownership over their own authorization handlers
+- flexibly supports additional sources of authorization (e.g. scoped user API keys) by defining
+  additional handlers
+
+#### Disadvantages
+
+- the interface works well for specific resources, but less well for bulk read operations (e.g.
```

Comment:
⛏️ Same comments. I would just capitalize all bullets and end them in periods as if they are sentences.